### PR TITLE
Story2 add collection

### PR DESCRIPTION
**Pull request by PNaing107**
Please can you review my Story 2? Thanks!

### CODE REVIEW

File: form.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    </p>
+    <form action="formValidation.php" method="post">
+        <label for="name">Name of Route</label><br>
+        <input type="text" maxlength="255" id="name" name="name" required><br>
+        <label for="country">Which Country is it in?</label><br>
+        <!-- for now this is a hardcoded list. In the future I should make this dynamic by getting Country list from db -->
+        <select name="country" id="country" required>
+            <?php
+            $db = connectToDB('epic-rides');
+            echo createOptionElement(getListOfCountries($db),'country_name');
```
Comment (nealioa) [inline form.php]: Add a space between `,` and `'`.

File: form.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            ?>
+        </select><br>
+        <label for="region">Which Region of the Country is it in?</label><br>
+        <select name="region" id="region" required>
+            <?php
+            $db = connectToDB('epic-rides');
+            echo createOptionElement(getListOfRegions($db),'region');
```
Comment (nealioa) [inline form.php]: Add a space between `,` and `'`.

File: formValidation.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// 1. Check Name of Route, URL, Alt Text & Short Description are strings, not empty and not larger than 255 characters
+$string_inputs = [
+    'name' => 'route name',
+    'url' => 'image url',
+    'alt_text' => 'image description',
+    'short_description' => 'route description'
+    ];
+
+$result = checkStringInput($string_inputs);
+if(!$result[0]) {
```
Comment (nealioa) [inline formValidation.php]: Add a space between `if` and `(`.

File: formValidation.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    echo '<p>Invalid ' . $result[1] .'. Make sure you provide a ' . $result[1] .' and it is no longer than 255 characters.</p>';
+    $validSubmission = false;
+}
+
+// 2. Check if valid url
+$result = validateURL($_POST['url']);
+if(!$result) {
```
Comment (nealioa) [inline formValidation.php]: Add a space between `if` and `(`.

File: formValidation.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+// 3. Execute addCollectionItem() if submitted data is valid
+if ($validSubmission) {
+    $db = connectToDB('epic-rides');
+    echo addCollectionItem($db);
+} else {
+    echo '<p>Please go back to the <a href="form.php">form</a> and try again.</p>';
+}
+
```
Comment (nealioa) [inline formValidation.php:34]: Unnecessary blank line. Delete it.

File: functions.php [hunk 6]
```diff
@@ -97,10 +100,156 @@ function displayCollection(array $array): string
...
+/**
+ * Function to insert form data into database
+ * @param PDO $pdo
+ * @return string
+ */
+function addCollectionItem(PDO $pdo): string
+{
+    // 1. Get location_id
+    $query = $pdo->prepare('SELECT `location_id` FROM `locations` WHERE `country_name`= :country_name AND `region`= :region;');
+    $query->bindParam(':country_name',$_POST['country']);
```
Comment (nealioa) [inline functions.php]: Add a space between `,` and `$_POST`.

File: functions.php [hunk 7]
```diff
@@ -97,10 +100,156 @@ function displayCollection(array $array): string
...
+    // 4. Get Route ID. Assuming INSERT into `routes` worked the largest `id` will be the `route_id`
+    // (this approach seems inefficient)
+    $query = $pdo->prepare('SELECT max(`id`) AS `id` FROM `routes`;');
+    $query->execute();
+    $result = $query->fetchAll();
+    (int) $route_id = $result[0]['id'];
+
+    // 5. Insert Statement into `images`
+    $query = $pdo->prepare(
+        'INSERT INTO `images` (`route_id`,`url`, `alt_text`)'.
```
Comment (nealioa) [inline functions.php]: Add a space between `'` and `.`.

**Review by nealioa — CHANGES_REQUESTED** [review]

File: form.php (hunk 1 above)
Comment (PNaing107) [reply form.php]: done

File: form.php (hunk 2 above)
Comment (PNaing107) [reply form.php]: done

File: formValidation.php (hunk 3 above)
Comment (PNaing107) [reply formValidation.php]: done

File: formValidation.php (hunk 4 above)
Comment (PNaing107) [reply formValidation.php]: done

File: formValidation.php (hunk 5 above)
Comment (PNaing107) [reply formValidation.php:34]: done

File: functions.php (hunk 6 above)
Comment (PNaing107) [reply functions.php]: done

File: functions.php (hunk 7 above)
Comment (PNaing107) [reply functions.php]: done

**Review by nealioa — APPROVED** [review]

**Pull request merged**